user-interface/src/main/java/life/qbic/datamanager/views/projects/create/CollaboratorsLayout.java

```diff
@@ -2,111 +2,99 @@
-        .withNullRepresentation(responsiblePersonField.getEmptyValue())
-        .withValidator(it -> responsiblePersonField.validate().isValid(), "")
-        .bind(bean -> bean.getResponsiblePerson().orElse(null),
-            ProjectCollaborators::setResponsiblePerson);
-
-    projectManagerField = new AutocompleteContactField("Project Manager", "Manager");
-    projectManagerField.setRequired(true);
-    collaboratorsBinder.forField(projectManagerField)
-        .withNullRepresentation(projectManagerField.getEmptyValue())
-        .withValidator(it -> projectManagerField.validate().isValid(), "")
-        .bind(ProjectCollaborators::getProjectManager,
-            ProjectCollaborators::setProjectManager);
 
+public class CollaboratorsLayout extends Div {
+
+  protected transient BoundContactField managerBinding;
+  protected transient BoundContactField investigatorBinding;
+  protected transient BoundContactField responsibleBinding;
+
+  public CollaboratorsLayout(PersonLookupService personLookupService) {
+
+    var fieldPrincipalInvestigator = createRequired("Principal Investigator", personLookupService);
+    var fieldProjectResponsible = createOptional(
+        "Project Responsible / Co-Investigator (optional)", personLookupService);
+    var fieldProjectManager = createRequired("Project Manager", personLookupService);
+    var currentUser = Utility.tryToLoadFromPrincipal().orElseThrow();
+    fieldProjectManager.setMyself(currentUser, "Set myself as project manager");
+    fieldProjectResponsible.setMyself(currentUser, "Set myself as project responsible");
+    fieldPrincipalInvestigator.setMyself(currentUser, "Set myself as principal investigator");
+    this.responsibleBinding = BoundContactField.createOptional(fieldProjectResponsible);
+    this.investigatorBinding = BoundContactField.createMandatory(fieldPrincipalInvestigator);
+    this.managerBinding = BoundContactField.createMandatory(fieldProjectManager);
     Span projectContactsTitle = new Span("Project Collaborators");
     projectContactsTitle.addClassName("title");
     Span projectContactsDescription = new Span(
         "Add the names and email address of the important contact people of the project.");
     addClassName("collaborators-layout");
     add(projectContactsTitle,
         projectContactsDescription,
-        principalInvestigatorField,
-        responsiblePersonField,
-        projectManagerField);
-    collaboratorsBinder.setValidatorsDisabled(false);
+        fieldPrincipalInvestigator,
+        fieldProjectResponsible,
+        fieldProjectManager);
   }
 
-  @Override
-  public String getDefaultErrorMessage() {
-    return "Please complete the mandatory information. Some input seems to be invalid.";
+  private static ContactField createRequired(String label,
+      PersonLookupService personLookupService) {
+    var contact = ContactField.createSimple(label, personLookupService);
+    contact.setRequiredIndicatorVisible(true);
```

Comment:
should be `setRequired(true)`. You are not in a composite field here but in some layouting class. What is the difference to the createMandatory of the bound contact field?

Reply:
While i agree that ideally there is the possiblity to just define this field as required, it's sadly not as simple since the validation was decoupled from the `ContactField` and moved into the` BoundContactField`. This means, that the `ContactField` itself does not provide a `setRequired()` method to be set to true but does have the UI indicator that the field has to filled. This is also the difference between this method and the createMandatory method of the `BoundContactField` which requires the `ContactField` as a parameter.